### PR TITLE
feature/51 - factor out attachFilters util

### LINKED ISSUE #51: factor out and export tree to filters fn
Right now in the course of running a search the contexture JSON is converted to appropriate elastic/mongo/etc stuff by running it through the providers here: https://github.com/smartprocure/contexture/blob/master/src/index.js#L28-L33

This logic is would be useful for those constructing advanced custom types as in some cases one has contexture search JSON to hand (e.g. from a saved search feature) and wants to use it to construct a `filter` for a node type (to make, for example, a filter that shows you things that don't match previous searches) - but this requires transforming the existing search to something provider-compliant itself as a node's `filter` is supposed to output a query fragment appropriate for the schema's provider.

Factoring the relevant lines and their dependencies into src/util and exporting appropriately will make this functionality available in custom node types

### PR DESCRIPTION
**Pull request by doug-patterson**
Fixes #51 

### Description
* factors out and exports the fn that attaches provider-constructed `_meta.filter` to the nodes in the search
* use case: custom node types need to be able to load a contexture blob and transform it into provider-appropriate filters in order to build filters that work by projecting contexture data (e.g. saved searches)

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 582df2d11bc0265aa46f23565e0409bd38535a1a
</p>

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "contexture",
-  "version": "0.10.0",
+  "version": "0.10.1",
```
Comment (ccalbaugh) [inline package.json]: I would consider this a minor version bump seeing as it is creating a new exported functionality

Comment (doug-patterson) [reply package.json]: seems fine to me, will update

**Review by ccalbaugh — APPROVED** [review]
Looks good

**Review by daedalus28 — APPROVED** [review]

**Pull request merged**